src/sql/platform/accounts/common/accountActions.ts

```diff
@@ -139,18 +132,16 @@ export class RefreshAccountAction extends Action {
 	) {
 		super(RefreshAccountAction.ID, RefreshAccountAction.LABEL, 'refresh-account-action codicon refresh');
 	}
-	public run(): Promise<boolean> {
+	public async run(): Promise<void> {
 		if (this.account) {
-			return Promise.resolve(this._accountManagementService.refreshAccount(this.account)
-				.then(() => true,
-					err => {
-						this.logService.error(`Error while refreshing account: ${err}`);
-						return Promise.reject(err);
-					}
-				));
+			try {
+				await this._accountManagementService.refreshAccount(this.account);
+			} catch (err) {
+				this.logService.error(`Error while refreshing account: ${err}`);
+			}
 		} else {
 			const errorMessage = localize('NoAccountToRefresh', "There is no account to refresh");
-			return Promise.reject(errorMessage);
+			throw new Error(errorMessage);
```

Comment:
Is this handled at the caller's side?

Reply:
Not sure, but the behavior is staying the same regardless (and this should never actually get here as well, we should always have an account for this action to refresh so if there isn't that's an error we can't really do anything about anyways)